playground-common/playground-plugin/src/main/kotlin/androidx/playground/SettingsParser.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.playground
+
+import java.io.File
+
+/**
+ * Helper class to parse the settings.gradle file from the main build and extract a list of
+ * projects.
+ */
+internal object SettingsParser {
+    /**
+     * Match lines that start with includeProject, followed by a require argument for project gradle
+     * path and an optional argument for project file path.
+     */
+    /* ktlint-disable max-line-length */
+    private val includeProjectPattern = Regex(
+        """^[\n\r\s]*includeProject\("(?<name>[a-z0-9-:]*)"(,[\n\r\s]*"(?<path>[a-z0-9-/]+))?.*\).*$""",
```

Comment:
I think \s includes \n and \r already. Also instead of IGNORE_CASE since you want `includeProject` to match exactly, you could use [a-zA-Z0-9] or if you don't like typing both a-z and A-Z you can use the POSIX character class `\p{alnum}`.